pkg/controller/master-controller-manager/application-secret-synchronizer/controller.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applicationsecretsynchronizer
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	k8scorev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName       = "application_secret_synchronization_controller"
+	secretTypeAnnotation = "apps.kubermatic.k8c.io/secret-type"
+	seedNS               = "kubermatic"
```

Comment:
You can use a seedGetter or a configGetter to determine the namespace dynamically, please do not hardcode this namespace into controllers.

Reply:
I am struggling a bit to get this done. So I know how to get the seedGetters and configGetter into my reconciler. However I do not understand how I can get the namespace the seed-controller-manager is running in out of a seedGetter or a configGetter.

While looking through other controllers that need to do the same, I realized they circumvent this by just passing in and empty namespace when creating objects (for example https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/master-controller-manager/cluster-template-synchronizer/controller.go#L121). I assume that the empty namespace then resolves to the seed-managers namespace, because its the default in the passed client. 

This would definitely work also for my case. However I think this is rather intransparent and a bit brittle; And for my curiosity: How can you get the seed-manager namespace out of a seedGetter of configGetter?

Reply:
> You can use a seedGetter or a configGetter to determine the namespace dynamically, please do not hardcode this namespace into controllers.

@xrstf I don't think seedGetter is required here. Adding a flag to make that namespace configurable should be enough

Reply:
> Adding a flag 

NOT a flag but passing `namespace` as a parameter to the `Add` method of the controller

Reply:
As to why we have a wild mix of namespaced and cluster-wide resources: History & RBAC.

Some of our resources are owners to others and Kubernetes doesn't allow cross-namespace owner refs. So this already necessitates some global resources.

I personally like namespaced resources and so I made the KubermaticConfiguration namespaced, even though we never fully 100% achieved the lofty goal of running multiple KKP installations in the same Kubernetes cluster (something we kind of somewhat a bit tried and did in the old CI cluster, before we used kind).

So even though it might look a bit chaotic, it's only slightly chaotic and if you tried to refactor it, you would notice the inter-dependencies between objects and that there's really no way to change any of that now :D